CorsixTH/Lua/app.lua

```diff
@@ -1680,57 +1680,131 @@ function App:loadLuaFolder(dir, no_results, append_to)
+  bug before the next milestone. The bug is then patched for the next milestone.
+  Moving forward, revisions should have a savegame_version bump of 1, as 10
+  savegame versions are reserved each release for patching.
+  All beta versions must have a savegame increment each time
+  Each patch note must begin with a '-'
+--]]
+local version_table = {
+  -- Format: major, minor, revision, patch (string), release date {YYYY, MM, DD}, savegame_version
+  {major = 0, minor = 0, revision = 8, patch = "", date = {2012, 01, 01}, version = 0}, -- Beta 8 or below
+  {major = 0, minor = 1, revision = 0, patch = "", date = {2012, 03, 24}, version = 51},
+  {major = 0, minor = 10, revision = 0, patch = "", date = {2012, 09, 24}, version = 53},
+  {major = 0, minor = 11, revision = 0, patch = "", date = {2012, 11, 17}, version = 54},
+  {major = 0, minor = 20, revision = 0, patch = "", date = {2013, 03, 24}, version = 66},
+  {major = 0, minor = 21, revision = 0, patch = "", date = {2013, 05, 24}, version = 72},
+  {major = 0, minor = 30, revision = 0, patch = "", date = {2013, 12, 24}, version = 78},
+  {major = 0, minor = 40, revision = 0, patch = "", date = {2014, 12, 22}, version = 91},
+  {major = 0, minor = 50, revision = 0, patch = "", date = {2015, 08, 31}, version = 105},
+  {major = 0, minor = 60, revision = 0, patch = "", date = {2016, 05, 30}, version = 111},
+  {major = 0, minor = 61, revision = 0, patch = "", date = {2017, 12, 21}, version = 122},
+  {major = 0, minor = 62, revision = 0, patch = "", date = {2018, 07, 21}, version = 127},
+  {major = 0, minor = 63, revision = 0, patch = "", date = {2019, 05, 24}, version = 134},
+  {major = 0, minor = 64, revision = 0, patch = "", date = {2020, 06, 16}, version = 138},
+  {major = 0, minor = 65, revision = 0, patch = "", date = {2021, 06, 19}, version = 156},
+  -- There was also 0.65.1, not differentiated by version number
+  {major = 0, minor = 66, revision = 0, patch = "", date = {2022, 06, 26}, version = 170},
+  {major = 0, minor = 67, revision = 0, patch = "", date = {2023, 08, 12}, version = 180},
+  {major = 0, minor = 68, revision = 0, patch = "", date = {2024, 10, 13}, version = 194},
+  {major = 0, minor = 69, revision = 0, patch = "-beta-1", date = {2025, 06, 07}, version = 216}
+}
+
+function App:getCurrentVersion()
+  return self.savegame_version
+end
+
+--! Requests data regarding a given savegame version
+--!param savegame_version (number) What to lookup, uses application version if blank
+--!param concatenate (boolean) (optional) Whether the version should be converted to a string
+--!return the matching version as a table, or as a string if concatenate is true.
+-- If no versions match it returns the base version with the savegame version
+-- attached e.g. 0.66.0-dev172
+function App:getVersion(savegame_version, concatenate)
+  savegame_version = savegame_version or self:getCurrentVersion()
+  local result
+  for i=#version_table, 1, -1 do
+    local version = version_table[i]
```

Comment:
Variable name is confusing. Likely, there are more such names.

It may make sense to reserve "version" for something numeric, ie `216` like the savegame version. Stuff like `1.2.3-something` looks like a release_number to me (release_name feels could work too, but not sure).

The "version_table" is not a list of versions, it has much more data. Perhaps `release[s]_table`?
If you still agree at this point, then the `version` here would be a `release` I guess.

Reply:
While on the subject of this, rename function maybe to `getReleaseData`?

Reply:
But it also produces a string every now and then.

Perhaps also make `getReleaseString`, which calls this function and then does the "concatenate" trick by itself?
The concatenate part has no influence on getting the data it seems, so concatenation is just a post-processing step.